plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/plan/CodeModernizerPlanEditor.kt

```diff
@@ -3,178 +3,492 @@
+
     override fun getName() = "CodeModernizerTransformationPlan"
+
     override fun getFile(): VirtualFile = virtualFile
+
     override fun setState(state: FileEditorState) {}
+
     override fun isModified() = false
+
     override fun isValid() = true
+
     override fun addPropertyChangeListener(listener: PropertyChangeListener) {}
+
     override fun removePropertyChangeListener(listener: PropertyChangeListener) {}
 
-    private fun title(text: String) = Panel().apply {
-        layout = FlowLayout(FlowLayout.LEFT)
-        val iconLabel = JLabel(AwsIcons.Logos.AWS_Q_GRADIENT)
-        val textLabel = JLabel(text).apply {
-            font = font.deriveFont(
-                CodeModernizerUIConstants.FONT_CONSTRAINTS.BOLD,
-                CodeModernizerUIConstants.PLAN_CONSTRAINTS.TITLE_FONT_SIZE
-            )
+    private fun title(text: String) =
+        Panel().apply {
+            layout = FlowLayout(FlowLayout.LEFT)
+            val iconLabel = JLabel(AwsIcons.Logos.AWS_Q_GRADIENT)
+            val textLabel =
+                JLabel(text).apply {
+                    font =
+                        font.deriveFont(
+                            CodeModernizerUIConstants.FONT_CONSTRAINTS.BOLD,
+                            CodeModernizerUIConstants.PLAN_CONSTRAINTS.TITLE_FONT_SIZE,
+                        )
+                }
+            add(iconLabel)
+            add(textLabel)
         }
-        add(iconLabel)
-        add(textLabel)
+
+    private fun createScrollPanel(): JPanel {
+        val arrowIcon = if (JBColor.isBright()) AwsIcons.CodeTransform.PLAN_ARROW_LIGHT else AwsIcons.CodeTransform.PLAN_ARROW_DARK
+        val scrollPanel = JPanel(BorderLayout())
+        val scrollIcon = JLabel(arrowIcon)
+        val scrollPane =
+            JEditorPane("text/html", message("codemodernizer.migration_plan.body.steps_scroll_top")).apply {
+                addHyperlinkListener { he ->
+                    if (he.eventType == HyperlinkEvent.EventType.ACTIVATED) {
+                        SwingUtilities.invokeLater { rootPanel.verticalScrollBar.setValue(0) }
+                    }
+                }
+                isEditable = false
+                isOpaque = false
+                alignmentX = Component.RIGHT_ALIGNMENT
+            }
+        scrollPanel.add(scrollPane, BorderLayout.CENTER)
+        scrollPanel.add(scrollIcon, BorderLayout.EAST)
+        return scrollPanel
     }
 
-    private fun transformationPlanPanel(plan: TransformationPlan) = JPanel(GridBagLayout()).apply {
-        val stepsIntroTitle = JLabel(message("codemodernizer.migration_plan.body.steps_intro_title")).apply {
-            font = font.deriveFont(
-                CodeModernizerUIConstants.FONT_CONSTRAINTS.BOLD,
-                CodeModernizerUIConstants.PLAN_CONSTRAINTS.TRANSFORMATION_STEP_TITLE_FONT_SIZE
-            )
-            border = CodeModernizerUIConstants.STEP_INTRO_TITLE_BORDER
+    private fun getFormattedString(s: String) =
```

Comment:
nit: can we rename this variable to be more descriptive?

Reply:
It's used in a couple of different places so hard to make it more specific while keeping it accurate